### PR TITLE
XWIKI-15381: Simple users don't have right to remove an annotation (t…

### PR DESCRIPTION
**Pull request by atallahade**
…hey created) from the Comments section

* Improve functional test.

I wanted to use JUnit 5 for the tests but a big refactor is needed. I'll leave that for XWiki 10.7.

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentsinline.vm [hunk 1]
```diff
@@ -141,7 +141,11 @@ $xwiki.ssfx.use('uicomponents/viewers/comments.css', true)
   #if ("$!commentSelection" != "")
     #set($isAnnotation = true)
   #end
-  #set ($isUserComment = $services.model.resolveDocument($comment.author, 'current') == $xcontext.userReference)
+  #if ($comment.author == '')
+    #set ($isUserComment = false)
+  #else
+    #set ($isUserComment = $services.model.resolveDocument($comment.author, 'user') == $xcontext.userReference)
+  #end
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentsinline.vm]: Could be written on a single line I think

    #set ($isUserComment = $comment.author != '' && $services.model.resolveDocument($comment.author, 'user') == $xcontext.userReference)

I would add a comment explaining that empty author means guest user and why we don't allow guest users to edit or delete annotations.

File: xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/CommentsTab.java [hunk 2]
```diff
@@ -173,4 +173,15 @@ public boolean hasEditButtonForCommentByID(int commentId)
...
+
+    /**
+     * @param commentId the comment id
+     * @return true if the comment has the delete button
+     * @since 10.6RC1
+     */
+    public boolean hasDeleteButtonForCommentByID(int commentId)
+    {
+        return getDriver().findElementsWithoutWaiting(By.xpath("//div[@id='xwikicomment_"
+            + commentId + "']//a[contains(@class, 'delete')]")).size() > 0;
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/CommentsTab.java]: I'm in favor of using the CSS selector when it's simpler

    "#xwikicomment_" + commentId + " a.delete"

File: xwiki-platform-core/xwiki-platform-annotations/xwiki-platform-annotation-test/xwiki-platform-annotation-test-pageobjects/src/main/java/org/xwiki/annotation/test/po/AnnotatableViewPage.java [hunk 3]
```diff
@@ -175,15 +184,28 @@ public void selectText(String annotationWord)
...
+
+    /**
+     * @param annotationId the annotation id.
+     * @return the comment id. Used by {@link org.xwiki.test.ui.po.CommentsTab}.
+     * @since 10.6RC1
+     */
+    public int getCommentId(String annotationId)
+    {
+        return Integer.valueOf(getDriver().findElementByXPath("//*[@id='_comments']" +
+            "//div[contains(@class, 'annotation')][a[contains(@href,'#" + annotationId + "')]]")
```
[the remark this answers is not in the document]
Comment (atallahade) [reply xwiki-platform-core/xwiki-platform-annotations/xwiki-platform-annotation-test/xwiki-platform-annotation-test-pageobjects/src/main/java/org/xwiki/annotation/test/po/AnnotatableViewPage.java]: I need to select the `.annotation` div and not the `a[href=...]` element. Can you write with a css selector "select the div which has a specific child"? I'm also not sure you can select the parent element of a selenium `WebElement`.

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentsinline.vm (hunk 1 above)
Comment (atallahade) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentsinline.vm]: Indeed.

File: xwiki-platform-core/xwiki-platform-annotations/xwiki-platform-annotation-test/xwiki-platform-annotation-test-pageobjects/src/main/java/org/xwiki/annotation/test/po/AnnotatableViewPage.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mflorea) [reply xwiki-platform-core/xwiki-platform-annotations/xwiki-platform-annotation-test/xwiki-platform-annotation-test-pageobjects/src/main/java/org/xwiki/annotation/test/po/AnnotatableViewPage.java]: You're right, I missed the ``[...]`` wrapping around the anchor.

**Pull request merged**